scripts/o3de/o3de/manifest.py

```diff
@@ -301,6 +301,83 @@ def get_engine_templates() -> list:
+    result = {gem_name: None for gem_name in gem_names_with_optional_gems}
+
+    engine_path = get_project_engine_path(project_path=project_path)
+    if not engine_path:
+        engine_path = get_this_engine_path()
+        if not engine_path:
+            logger.error('Failed to find an engine path for the project at '
+                            f'"{project_path}" which is required to resolve gem dependencies.')
+            return result
+
+        logger.warning('Failed to determine the correct engine for the project at '
+                        f'"{project_path}", falling back to this engine at {engine_path}.')
+    
+    engine_json_data = get_engine_json_data(engine_path=engine_path)
+    if not engine_json_data:
+        logger.error('Failed to retrieve engine json data for the engine at '
+                     f'"{engine_path}" which is required to resolve gem dependencies.')
+        return result 
+
+    all_gems_json_data = get_gems_json_data_by_name(engine_path=engine_path, 
+                                                    project_path=project_path, 
+                                                    include_manifest_gems=True, 
+                                                    include_engine_gems=True)
+
+    # we need a mapping of gem name to gem name with version specifier because
+    # the resolver will remove the version specifier
+    gem_names_with_version_specifiers = {}
+    for gem_name_with_specifier in gem_names_with_optional_gems:
+        gem_name_only, _ = utils.get_object_name_and_optional_version_specifier(gem_name_with_specifier)
+        gem_names_with_version_specifiers[gem_name_only] = gem_name_with_specifier
+
+    # Try to resolve with optional gems
+    resolved_gems, errors = compatibility.resolve_gem_dependencies(gem_names_with_optional_gems, 
+                                                             all_gems_json_data, 
+                                                             engine_json_data, 
+                                                             include_optional=True)
+    if errors:
+        # Try without optional gems
+        gem_names_without_optional = utils.get_gem_names_set(active_gem_names, include_optional=False)
+        resolved_gems, errors = compatibility.resolve_gem_dependencies(gem_names_without_optional, 
+                                                                 all_gems_json_data, 
+                                                                 engine_json_data,
+                                                                 include_optional=False)
+    if not errors:
+        for _, gem in resolved_gems.items():
+            gem_name = gem.gem_json_data['gem_name']
+            gem_name_with_specifier = gem_names_with_version_specifiers.get(gem_name,gem_name)
+            if gem_name_with_specifier in result or include_dependencies:
+                result[gem_name_with_specifier] = gem.gem_json_data['path'].resolve().as_posix()
```

Comment:
Uh, isn't this path relative to the gem directory containing the "external_subdirectory" entry for this sub gem?
Wouldn't this result in the path being resolved against the current working directory therefore resulting in the wrong path?

Reply:
I probably don't need to call resolve here,  the paths should already be absolute at this point. 
The 'path' comes from `get_gems_json_data_by_name` which pulls the paths from 
`get_manifest_external_subdirectories`, `get_project_external_subdirectories`, and then uses `get_gems_from_external_subdirectories`.
It appears that all the functions return absolute paths by appending the 'external_subdirectory' except `get_manifest_external_subdirectories`

It seems to be working fine for all the Atom sub gems so far, but couldn't hurt to try the MPS gems too.

Reply:
I verified with MPS that all paths are absolute for gems in the engine and manifest including subdir gems like those in `Atom` and `MPSGems`